### ISSUE TITLE
update > 1.5.0.6 not working

### ISSUE THREAD

**Issue by fguichard09** [baskren/Forms9Patch#3]
Hello,

I'm now using Forms9Patch 1.5.0.6 with SkiaSharp 1.60.0 in a Xamarin.Forms project.
If I update Forms9Patch to the latest version (1.6.4 with SkiaSharp 1.59.3), I randomly get the following crash :

`[MonoDroid] System.NullReferenceException: Object reference not set to an instance of an object.
[MonoDroid]   at Xamarin.Forms.BindableObject.GetContext (Xamarin.Forms.BindableProperty property) [0x00013] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:534 
[MonoDroid]   at Xamarin.Forms.BindableObject.GetOrCreateContext (Xamarin.Forms.BindableProperty property) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:544 
[MonoDroid]   at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x000ba] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:393 
[MonoDroid]   at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle, System.Boolean checkAccess) [0x0003d] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:573 
[MonoDroid]   at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:99 
[MonoDroid]   at Forms9Patch.Label.set_IsDynamicallySized (System.Boolean value) [0x00000] in C:\Users\ben\Development\Forms9Patch2\Forms9Patch\Forms9Patch.Source\Elements\Label.cs:152 
[MonoDroid]   at Forms9Patch.Label.InvalidateMeasure () [0x00007] in C:\Users\ben\Development\Forms9Patch2\Forms9Patch\Forms9Patch.Source\Elements\Label.cs:338 
[MonoDroid]   at Forms9Patch.Label.OnPropertyChanged (System.String propertyName) [0x00211] in C:\Users\ben\Development\Forms9Patch2\Forms9Patch\Forms9Patch.Source\Elements\Label.cs:325 
[MonoDroid]   at Forms9Patch.Label+<>c__DisplayClass60_0.<OnPropertyChanged>b__0 () [0x00000] in C:\Users\ben\Development\Forms9Patch2\Forms9Patch\Forms9Patch.Source\Elements\Label.cs:268 
[MonoDroid]   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <289faaef20de42089854c8bef59604fd>:0 
[MonoDroid]   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <289faaef20de42089854c8bef59604fd>:0 
[MonoDroid]   at (wrapper dynamic-method) System.Object.44(intptr,intptr)`

All other packages are up to date.

I also tried to update to every versions above 1.5.0.6 but I get the same result.
I get the exact same bug wether on Android or iOS.

Thanks for your help.

**Comment by baskren** [comment]
Looking at the stack trace, there is not enough information to identify the root cause of this issue.  In order for me to fix this, I will need a small demo project (preferably in a GitHub or BitBucket repo) that demonstrates this issue.

Ben Askren
BuildCalc developer
ben@buildcalc.com

 <http://buildcalc.com/>
 <http://buildcalc.com/>

**Comment by fguichard09** [comment]
Unfortunately, I cannot reduce my project because the crash is random and the less code I have, the less frequently the bug appears.

I found out the line causing the bug (the last one below) :

```
StackLayout layout = new StackLayout
{
	Orientation = StackOrientation.Horizontal
};

var name = new Forms9Patch.Label()
{
	FontSize = MyProject.Constants.FONTSIZE_NORMAL,
	VerticalOptions = LayoutOptions.Center,
	LineBreakMode = LineBreakMode.WordWrap,
	HtmlText = txt
};
switchLayout.Children.Add(name);
```

The error message is either 

> object reference not set (as shown above)
or
> Destination array is not long enough to copy all the items in the collection. Check array index and length.

If I use a version <= 1.5.0.6 that's ok. If I use a version >1.5.0.6, the crash appears. If I replace Forms9Patch.Label with Label (and the property 'HtmlText' with 'Text'), it's ok.
I tried to make a loop within a brand new shared library project but no luck.

**Comment by baskren** [comment]
I am sorry but there is not enough information.  Perhaps, given that your issue occurs with setting the HtmlText but not Text property, you could at least share the string you are passing to HtmlText?

**Comment by fguichard09** [comment]
I tried to set Text=""  in order for HtmlText not to be overloaded, but no change.
A different string will generate the crash each time the project is launched. So I don't think the problem is the text itself. One text I could log before the crash was "Variation de section entre Ø CdR = 37 et Ø CF = 321".
Thanks for your help anyway. I'll keep this version and test at regular intervals wether this problem has been solved.
